src/ddscxx/include/org/eclipse/cyclonedds/topic/datatopic.hpp

```diff
@@ -476,21 +468,13 @@ void sertopic_realloc_samples(
    * be invoked by ddsi_sertopic_alloc_sample, in which case oldCount is always 0,
    * count is always 1 and the old pointer is always null.
    */
-  assert(oldcount == 0);
-  assert(count == 1);
-  assert(old == nullptr);
-
   ptrs[0] = new T();
 }
 
 template <typename T>
 void sertopic_free_samples(
-  const struct ddsi_sertopic* d, void** ptrs, size_t count,
-  dds_free_op_t op)
+  const struct ddsi_sertopic*, void** ptrs, size_t, dds_free_op_t op)
 {
```

Comment:
the assert(count == 1) several lines down should also be removed, since it is no longer one of the function parameters